MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -446,7 +455,38 @@ - (nullable NSURL *)buildTokenRequestURLWithAppSecret:(NSString *)appSecret rele
   return components.URL;
 }
 
-- (void)openURLInEmbeddedSafari:(NSURL *)url fromClass:(Class)clazz {
+- (void)openURLInAuthenticationSessionWith:(NSURL *)url fromClass:(Class)clazz {
+  MSLogDebug([MSDistribute logTag], @"Using SFAuthenticationSession to open URL: %@", url);
+
+  NSString *callbackUrlScheme = [NSString stringWithFormat:kMSDefaultCustomSchemeFormat, self.appSecret];
+
+  if (@available(iOS 11.0, *)) {
```

Comment:
Neat :)

Reply:
Problem is that this requires Xcode 9 and Xcode 9 breaks our tests for MSDistribute (not related to this PR)